### PR TITLE
Prevent saving if the player is extinct

### LINKED ISSUE #2593: Saving if the player is extinct should be prevented
just tested and I was able to make a quicksave when the player is dead.

Two things:
- [ ] prevent quick save when extinct
- [ ] prevent save from the pause menu (maybe extinction could send a signal to the pause menu telling it to disable the save option permanently? starting a new game or loading a save should restore that just fine)

### PR DESCRIPTION
**Pull request by hhyyrylainen**
**Brief Description of What This PR Does**

prevent saving if extinct

**Related Issues**

closes #2593

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/saving/InProgressSave.cs [hunk 1]
```diff
@@ -30,8 +30,14 @@ public class InProgressSave : IDisposable
 
     private bool success;
     private string message;
+
+    /// <summary>
+    ///   Failure exception or description. Depends on exceptionIsException
+    /// </summary>
     private string exception;
 
+    private bool exceptionIsException = true;
```
Comment (FuJa0815) [inline src/saving/InProgressSave.cs:39]: That variable sounds very... strange. Maybe rename the `exception` variable?

File: src/saving/SaveStatusOverlay.cs [hunk 2]
```diff
@@ -80,15 +85,19 @@ public void ShowMessage(string message, float visibleTime = 0.7f)
     /// <param name="message">Message to show</param>
     /// <param name="exception">Extra / exception info to show</param>
     /// <param name="returnToMenu">
-    ///   If true closing the dialog returns to menu. If false the dialog is just closed (and game is unpaused)
+    ///    If true closing the dialog returns to menu. If false the dialog is just closed (and game is unpaused)
```
Comment (FuJa0815) [inline src/saving/SaveStatusOverlay.cs:88]: Indentation wrong

File: locale/en.po [hunk 3]
```diff
@@ -3070,11 +3070,19 @@ msgstr "Overwrite existing save:"
 msgid "OVERWRITE_EXISTING_SAVE_TITLE"
 msgstr "Overwrite existing save?"
 
-#: ../src/saving/SaveHelper.cs:379
+#: ../src/saving/SaveHelper.cs:382
+msgid "SAVING_NOT_POSSIBLE"
+msgstr "Saving is not currently possible due to:"
```
Comment (FuJa0815) [inline locale/en.po:3075]:
```suggestion
msgstr "Saving is currently not possible due to:"
```

**Review by FuJa0815 — DISMISSED** [review]
In-game testing pending

File: src/saving/InProgressSave.cs (hunk 1 above)
Comment (hhyyrylainen) [reply src/saving/InProgressSave.cs]: I did try, couldn't really think of anything better than what I ended up with.

File: locale/en.po (hunk 3 above)
Comment (hhyyrylainen) [reply locale/en.po:3075]: I found this: https://english.stackexchange.com/questions/5464/currently-not-or-not-currently
which says that both ways are correct. Though, having "not" first emphasises the "not", which I don't think is too much of a difference either way, so... I'm not convinced it's worth the minuscule trouble of changing this.

Comment (hhyyrylainen) [reply locale/en.po:3075]: Would you consider that a fine explanation? If you can get people to reply on discord in the #translation channel or maybe on the community forums, this'll be pretty easy to change on https://translate.revolutionarygamesstudio.com/

Comment (FuJa0815) [reply locale/en.po:3075]: Nah, should be fine. I've talked with somebody who is better in english than I am, and he said that both are acceptable. No need to change it.

Comment (hhyyrylainen) [reply locale/en.po:3075]: Okay that's fine. But as I linked that stackexchange answer it says that there is a slight emphasis difference, which we can discuss in the future if anyone cares enough to want to change it.

**Pull request merged**